workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,139 @@
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in reserveEntityIdsResponseData.CommandResponses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnReserveEntityIdsResponseInternal(receivedResponse);
+                    }
+                }
+            }
+
+            for (var i = 0; i < createEntityResponseData.Length; ++i)
+            {
+                var entity = createEntityResponseData.Entities[i];
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in createEntityResponseData.CommandResponses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnCreateEntityResponseInternal(receivedResponse);
+                    }
+                }
+            }
+
+            for (var i = 0; i < deleteEntityResponseData.Length; ++i)
+            {
+                var entity = deleteEntityResponseData.Entities[i];
+                var worldCommandResponseHandlers = GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in deleteEntityResponseData.CommandResponses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnDeleteEntityResponseInternal(receivedResponse);
+                    }
+                }
+            }
+        }
+
+        private WorldCommands.Requirables.WorldCommandResponseHandler[] GetWorldCommandResponseHandlersForEntity(
+            Entity entity)
+        {
+            var entityToReaderWriterStore = gameObjectDispatcherSystem.entityToReaderWriterStore;
+
+            if (!entityToReaderWriterStore.TryGetValue(entity, out var injectableStore))
+            {
+                return null;
+            }
+
+            if (!injectableStore.TryGetInjectablesForComponent(WorldCommandResponseHandlerInjectableId,
+                out var injectables))
+            {
+                return null;
+            }
+
+            if (injectables.Count == 0)
+            {
+                return null;
+            }
+
+            return Array.ConvertAll(injectables.ToArray(),
```

Comment:
This is not oke in a hot loop

Reply:
Offline discussion: let's do the optimisation in next PR